### PR TITLE
csproj and vs 2017 updates to ef mvc core tutorial

### PR DESCRIPTION
**Pull request by tdykstra**
@rowanmiller @Rick-Anderson please review
Aside from general project.json to .csproj and VS 2017 changes, this also updates to EF Core 1.1.  

Links to staging for sections with significant changes:

* In the intro tutorial: the [section about adding EF support to a project](https://review.docs.microsoft.com/en-us/aspnet/core/data/ef-mvc/intro?branch=efmvc#entity-framework-core-nuget-packages) (.csproj instead of project.json)
* In the migrations tutorial: [the section about PMC vs CLI](https://review.docs.microsoft.com/en-us/aspnet/core/data/ef-mvc/migrations?branch=efmvc#command-line-interface-cli-vs-package-manager-console-pmc) (no longer warns that PMC is unstable)
* In the reading related data tutorial: 
  * [the intro section that explains eager vs explicit loading](https://review.docs.microsoft.com/en-us/aspnet/core/data/ef-mvc/read-related-data?branch=efmvc#eager-explicit-and-lazy-loading-of-related-data)
  * [the code for explicit loading](https://review.docs.microsoft.com/en-us/aspnet/core/data/ef-mvc/read-related-data?branch=efmvc#explicit-loading) (now using the same code from the EF 6 version of the tutorial)
* In the concurrency tutorial: [the code for getting database values](https://review.docs.microsoft.com/en-us/aspnet/core/data/ef-mvc/concurrency?branch=efmvc#update-the-edit-methods-in-the-departments-controller) (now using the same code from the EF 6 version of the tutorial)

### CODE REVIEW

File: aspnetcore/data/ef-mvc/intro.md [hunk 1]
```diff
@@ -102,13 +102,19 @@ Because you used the **Individual User Accounts** option when you created the pr
 
 If you want to add EF Core support to a new project that you create without the **Individual User Accounts** option, install the following NuGet packages:
 
-* The package for the database provider you want to target. To use SQL Server, the package is [Microsoft.EntityFrameworkCore.SqlServer](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer/). For a list of available providers see [Database Providers](https://docs.microsoft.com/ef/core/providers/).
+* The package for the database provider that you want to target. For SQL Server, the package is [Microsoft.EntityFrameworkCore.SqlServer](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer/). For a list of available providers see [Database Providers](https://docs.microsoft.com/ef/core/providers/).
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/intro.md]: ~~The package for~~ the database provider that you want to target.

File: aspnetcore/data/ef-mvc/intro.md [hunk 2]
```diff
@@ -102,13 +102,19 @@ Because you used the **Individual User Accounts** option when you created the pr
...
+
+* The package for running EF command-line tools in **Package Manager Console**:[Microsoft.EntityFrameworkCore.Tools](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Tools).
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/intro.md]: The EF command-line tools for PM

File: aspnetcore/data/ef-mvc/intro.md [hunk 3]
```diff
@@ -102,13 +102,19 @@ Because you used the **Individual User Accounts** option when you created the pr
 
 If you want to add EF Core support to a new project that you create without the **Individual User Accounts** option, install the following NuGet packages:
 
-* The package for the database provider you want to target. To use SQL Server, the package is [Microsoft.EntityFrameworkCore.SqlServer](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer/). For a list of available providers see [Database Providers](https://docs.microsoft.com/ef/core/providers/).
+* The package for the database provider that you want to target. For SQL Server, the package is [Microsoft.EntityFrameworkCore.SqlServer](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer/). For a list of available providers see [Database Providers](https://docs.microsoft.com/ef/core/providers/).
+
+* The package for running EF command-line tools in **Package Manager Console**:[Microsoft.EntityFrameworkCore.Tools](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Tools). 
+
+* The package for running EF command-line tools at a command prompt: [Microsoft.EntityFrameworkCore.Tools.DotNet](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Tools.DotNet).
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/intro.md]: ditto ~~The package for~~

File: aspnetcore/data/ef-mvc/intro.md [hunk 4]
```diff
@@ -102,13 +102,19 @@ Because you used the **Individual User Accounts** option when you created the pr
...
+
+* The package for running EF command-line tools in **Package Manager Console**:[Microsoft.EntityFrameworkCore.Tools](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Tools). 
+
+* The package for running EF command-line tools at a command prompt: [Microsoft.EntityFrameworkCore.Tools.DotNet](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Tools.DotNet). 
 
-* The package for the EF command-line tools:   [Microsoft.EntityFrameworkCore.Tools](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Tools/1.0.0-preview2-final). This package is a preview release, so to install it you have to enable preview release installation. After installing the package, you also have to add a reference to it in the `tools` collection in the *project.json* file.
+  This package is installed by adding it to the `DotNetCliToolReference` collection in the *.csproj* file, as shown.
 
-If you open the *project.json* file, you'll see that these packages are already installed.
+  [!code-xml[](intro/samples/cu/ContosoUniversity.csproj?range=30-34&highlight=2)]
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/intro.md]: We know customers will copy/paste and soon that version # will be out of date. But I think that's probably the least of the evils. Probably better to not have a note: These version numbers were correct at the time of writing. You generally want to download the latest stable version.

File: aspnetcore/data/ef-mvc/intro.md [hunk 5]
```diff
@@ -102,13 +102,19 @@ Because you used the **Individual User Accounts** option when you created the pr
...
+  
+ (You can edit the *.csproj* file by right-clicking the project name in **Solution Explorer** and selecting **Edit ContosoUniversity.csproj** from the context menu.)
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/intro.md]: Nice- we should standardize on this format. With *project.json* the name was constant, now the file name changes with each project. Should we consider the less correct/complete but more terse (and probably good enough)

(You can edit the *.csproj* file by right-clicking the project name in **Solution Explorer** and selecting **Edit ContosoUniversity.csproj** ~~from the context menu~~.)

File: aspnetcore/data/ef-mvc/intro.md [hunk 6]
```diff
@@ -124,6 +130,8 @@ In the following sections you'll create a class for each one of these entities.
 
 ![Student entity diagram](intro/_static/student-entity.png)
 
+In the project folder, create a *Models* folder.
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/intro.md]: Bummer, the tooling doesn't create these folders?

File: aspnetcore/data/ef-mvc/intro.md [hunk 7]
```diff
@@ -63,16 +63,16 @@ Open Visual Studio 2015 and create a new ASP.NET Core C# web project named "Cont
...
-* Clear the **Host in the cloud** check box.
+* Set **Authentication** to **Individual User Accounts**.
 
 * Click **OK**
 
   ![New ASP.NET Project dialog](intro/_static/new-aspnet.png)
 
 > [!NOTE]
-> Don't miss setting authentication to **Individual User Accounts**. You won't be using authentication in this tutorial, but you need to enable it because of a limitation of .NET Core Preview 2 Visual Studio tooling. Scaffolding for MVC controllers and views only works when **Individual User Accounts** authentication is enabled.
+> Make sure the **Authentication** setting is **Individual User Accounts**. You won't be using authentication in this tutorial, but this setting automates some set-up tasks that you'd have to do manually otherwise.
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/intro.md]: Are you saying it works without auth enabled? When we have time we should really update our tooling to not use Auth as it adds so much unnecessary code to the project.

File: aspnetcore/data/ef-mvc/advanced.md [hunk 8]
```diff
@@ -1,8 +1,8 @@
 ---
 title: ASP.NET Core MVC with EF Core - Advanced - 10 of 10 | Microsoft Docs
 author: tdykstra
-description: 
-keywords: ASP.NET Core,
+description: This tutorial introduces several topics that are useful to be aware of when you go beyond the basics of developing ASP.NET web applications that use Entity Framework Core.
+keywords: ASP.NET Core, Entity Framework Core, raw sql, examine sql, repository pattern, unit of work pattern, automatic change detection, existing database
 ms.author: tdykstra
 manager: wpickett
 ms.date: 10/14/2016
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/advanced.md]: update date

File: aspnetcore/data/ef-mvc/concurrency.md [hunk 9]
```diff
@@ -151,37 +151,31 @@ The code in the catch block for that exception gets the affected Department enti
 
 [!code-csharp[Main](intro/samples/cu/Controllers/DepartmentsController.cs?range=164)]
 
-The `Entries` collection will have just one `EntityEntry`, object, and that object has the new values entered by the user.
+The `Entries` collection will have just one `EntityEntry` object, and you can use that object to get the new values entered by the user abd the current database values.
```
Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/concurrency.md]: consider. 
The `Entries` collection will have just one `EntityEntry` object.  You

Comment (Rick-Anderson) [inline aspnetcore/data/ef-mvc/concurrency.md]: entered by the user **and** the current database values.

File: aspnetcore/data/ef-mvc/intro.md (hunk 7 above)
Comment (tdykstra) [inline aspnetcore/data/ef-mvc/intro.md]: Yes scaffolding works without auth, and we can update the tutorial to not use auth later.

File: aspnetcore/data/ef-mvc/intro.md (hunk 6 above)
Comment (tdykstra) [inline aspnetcore/data/ef-mvc/intro.md]: My mistake - on my first pass I was planning to rewrite the tutorial to select no auth, then I switched back to auth when I realized how much work the change was going to require, but missed removing this comment.

**Comment by Rick-Anderson** [comment]
@tdykstra  we can remove auth when we have time.

**Pull request merged**